### PR TITLE
Use MinVer

### PR DESCRIPTION
**Pull request by sungam3r**
See https://github.com/adamralph/minver

It allows to not bother with setting version manually in code/csproj. I tested it in 4 repos. So far it works.

### CODE REVIEW

**Comment by sungam3r** [comment]
https://stackoverflow.com/questions/75537987/how-do-i-get-the-github-actions-section-in-the-visual-studio-solution-view

File: src/Directory.Build.props [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup>
-    <VersionPrefix>7.3.0-preview</VersionPrefix>
+    <MinVerDefaultPreReleaseIdentifiers>preview</MinVerDefaultPreReleaseIdentifiers>
```
Comment (sungam3r) [inline src/Directory.Build.props:4]: https://github.com/adamralph/minver#can-i-change-the-default-pre-release-identifiers-from-alpha0-to-something-else

File: .github/workflows/build-artifacts-code.yml [hunk 2]
```diff
@@ -21,6 +21,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (sungam3r) [inline .github/workflows/build-artifacts-code.yml:25]: MinVer need tags to work. It's a known issue with `actions/checkout` - this action has no explicit option to fetch tags. `fetch-depth: 0` effectivelty fetches all tags. Yep, it fetches all commits as well 😞 .

**Comment by Shane32** [comment]
I don't favor this.  First, I don't like having to retrieve the entire commit history, further delaying test runs.  But even more importantly, I don't like introducing another build-time dependency when it is not necessary.  We already have too many dependencies here that constantly "need" updating.  It also makes it more difficult to understand the versioning process by looking at the csproj and workflow files, whereas now it is clear and precise.

What are the benefits?  The numbering for dev builds is more "accurate"?  Does not matter; they require quite a lot of work to use, and the few users there are (if any) should not care.

So in summary, I see this as something that increases, not decreases, the maintenance required for GraphQL.NET.  As an open-source project maintained by only two of us, both of which are extremely pressed for time, I cannot endorse it.

**Comment by Shane32** [comment]
Another option is to leave the version in directory.build.props at 20.0.0-preview. It will always generate a newer version than the latest published version. (This only matters if using another published dependency in conjunction with a dev build.)

**Comment by Shane32** [comment]
If I have time perhaps I will test MinVer in one of my own repos

**Comment by sungam3r** [comment]
If I would choose between some kind of fictitious version and the absense of version in sources, then I will prefer the latter.

**Comment by sungam3r** [comment]
> If I have time perhaps I will test MinVer in one of my own repos

👍 Honestly it took me a couple of **years** to finally try, I have long known about this package and saw its use in other famous repositories.

File: .github/workflows/build-artifacts-code.yml (hunk 2 above)
Comment (sungam3r) [reply .github/workflows/build-artifacts-code.yml:26]: fixed in https://github.com/actions/checkout/pull/579

**Comment by Shane32** [comment]
I think what we have already works perfectly.

**Comment by Shane32** [comment]
I'll close it for now then.  We can reopen if we wish to discuss further.
